http/digest/src/test/java/org/wildfly/security/http/digest/DigestMechanismFactoryTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.http.digest;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.wildfly.security.http.HttpAuthenticationException;
+import org.wildfly.security.http.HttpServerAuthenticationMechanism;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import java.io.IOException;
+import java.util.HashMap;
+
+/**
+ * Tests of DigestMechanismFactory Class.
+ *
+ * @author Keshav Kumar
+ */
+
+public class DigestMechanismFactoryTest {
+
+    private DigestMechanismFactory digestMechanismFactory = new DigestMechanismFactory();
+
+    CallbackHandler dummyCallbackHandler = new CallbackHandler() {
+        @Override
+        public void handle(Callback[] callbacks) throws IOException, UnsupportedCallbackException {
+        }
+    };
+
+    /**
+     * Tests that getMechanismNames(Map) correctly
+     * handles null or empty properties parameter as possible value.
+     */
+    @Test
+    public void testGetMechanismNamesPropertiesNull(){
+        digestMechanismFactory.getMechanismNames(null);
```

Comment:
Would be good to check the return value here and below.

Reply:
Thank you for reviewing the PR @fjuma .I have updated the PR.Please have a look.